### DISCUSSION TITLE
Is there a way to disable sorting by default but enable in specific columns?

### DISCUSSION THREAD

**Discussion by edelfrade** [TanStack/table#2490] [General]
`defaultCanSort` only works to force enable sorting on columns without an accessor.
I've been able to `disableSortBy: true` at the table level, but enabling it at the column level does not overwrite it. 

The plan is allow sorting in specific columns without having to declare it each time a column is created. 

Currently setting up a default column objet that will extend each of the columns created but was wondering if there was some default implementation I'm missing. 

Thank you for your time.

**Comment by boldwade** [comment]
@ernerock , did you ever find the answer? I actually have the opposite question. I would like to prevent sorting/filtering on one or two columns.

**Reply by gunnartorfis** [reply]
@boldwade Add disableFilters: true, to the respective column object.

**Reply by edelfrade** [reply]
@boldwade I ended up having my table component `disableSortBy: true` by default, then if a specific table was sortable I would pass a prop to enable sorting and then as @gunnartorfis mentioned add `disableSortBy: true` to each non-sortable column

**Reply by ziverlichson** [reply]
You can add the following column options to disable sorting/filtering on specific columns:
disableSortBy: true- for sorting
disableFilters: false - for filtering
Check [here ](https://react-table-omega.vercel.app/docs/api/useSortBy#column-options) for more details regarding disableSortBy and [here ](https://react-table-omega.vercel.app/docs/api/useFilters#column-options) for more details regarding disableFilters

**Reply by muhammed-salman** [reply]
I have the same question, I have to disable sorting on 1 or 2 column(s) and in rest of the columns it should be enabled. Passing `defaultCanSort: true` in table options enables the sorting on all columns, however I'm not able to disable sorting by passing `defaultCanFilter: false` or `disableSortBy: true` in a particular column option, what I'm doing wrong?

**Reply by tannerlinsley** [reply]
@muhammed-salman If you are attempting to do this in v7, the override logic is a bit messy. This works as expected in v8.

**Reply by nathan-isaac** [reply]
It looks like the code defaults sorting to `true`. It would be nice to be able to configure this value for the cases where I only want to sort 1 or 2 columns out of n number of columns. 

https://github.com/TanStack/table/blob/main/packages/table-core/src/features/Sorting.ts#L310

**Comment by SurajanShrestha** [comment]
@edelfrade @tannerlinsley @boldwade @muhammed-salman  
What i do is pass a custom property in that column object which i need to sort.

```
const columns = useMemo(
    () => [
      {
        Header: 'Title',
        accessor: 'title',
        isSortable: true, // This is my custom property
      },
      {
        Header: 'Type',
        accessor: 'type',
      },
    ],
    []
  );
```
And in my Table component, i do this:

`<th {...column.getHeaderProps(column?.isSortable ? column.getSortByToggleProps() : {})}>`
